core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -153,6 +153,8 @@ public static void merge(List<PathDetail> pathDetails, List<PathDetail> otherDet
         // Make sure that the PathDetail list is merged correctly at via points
         if (!pathDetails.isEmpty() && !otherDetails.isEmpty()) {
             PathDetail lastDetail = pathDetails.get(pathDetails.size() - 1);
+            // Add +1 for the via point
+            lastDetail.setLast(lastDetail.getLast()+1);
```

Comment:
Hmmh, this feels wrong and we should probably skip adding the via points opposed to my recent comment in #1139?